samples/snippets/sample_templates.py

```diff
@@ -0,0 +1,195 @@
+
+    Returns:
+        InstanceTemplate object containing information about given template.
+    """
+    template_client = compute_v1.InstanceTemplatesClient()
+    return template_client.get(project=project_id, instance_template=template_name)
+
+
+def list_instance_templates(project_id: str) -> Iterable[compute_v1.InstanceTemplate]:
+    """
+    Get a list of InstanceTemplate objects available in given project.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you use.
+
+    Returns:
+        Iterable list of InstanceTemplate objects.
+    """
+    template_client = compute_v1.InstanceTemplatesClient()
+    return template_client.list(project=project_id)
+
+
+def create_template(project_id: str, template_name: str) -> compute_v1.InstanceTemplate:
+    """
+    Create a new template in given project with provided name.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you use.
+        template_name: name of the new template that is created.
+
+    Returns:
+        InstanceTemplate object describing the new instance template.
+    """
+    # Describe the size and source image of the boot disk to attach to the instance.
+    disk = compute_v1.AttachedDisk()
+    initialize_params = compute_v1.AttachedDiskInitializeParams()
+    initialize_params.source_image = (
+        "projects/debian-cloud/global/images/family/debian-11"
+    )
+    initialize_params.disk_size_gb = 250
+    disk.initialize_params = initialize_params
+    disk.auto_delete = True
+    disk.boot = True
+
+    # Use the default network, without specifying a subnetwork.
+    network_interface = compute_v1.NetworkInterface()
+    network_interface.name = "global/networks/default"
+
+    # Let the instance have external IP
+    network_interface.access_configs = [compute_v1.AccessConfig()]
+    network_interface.access_configs[0].name = "External NAT"
+    network_interface.access_configs[0].type_ = compute_v1.AccessConfig.Type.ONE_TO_ONE_NAT
+    network_interface.access_configs[0].network_tier = (
+        compute_v1.AccessConfig.NetworkTier.PREMIUM
+    )
+
+    template = compute_v1.InstanceTemplate()
+    template.name = template_name
+    template.properties.disks = [disk]
+    template.properties.machine_type = "e2-standard-4"
+    template.properties.network_interfaces = [network_interface]
+
+    template_client = compute_v1.InstanceTemplatesClient()
+    operation_client = compute_v1.GlobalOperationsClient()
+    op = template_client.insert(project=project_id, instance_template_resource=template)
+    operation_client.wait(project=project_id, operation=op.name)
+
+    return template_client.get(project=project_id, instance_template=template_name)
+
+
+def create_template_from_instance(project_id: str, instance: str, template_name: str) -> compute_v1.InstanceTemplate:
+    """
+    Create a new instance template based on an existing instance.
```

Comment:
Given the content of this function, should it include some mention of changing values?

For example:
"Create a new instance template based on an existing instance. This new template specifies a different boot disk."